src/server/pa_rot.rs

```diff
@@ -58,150 +34,42 @@ where
-                })
-            })
-            .handle::<protocol::DeviceInfo, _>(|zelf, _| {
-                Ok(protocol::device_info::DeviceInfoResponse {
-                    info: zelf.opts.identity.unique_device_identity(),
-                })
-            })
-            .handle::<protocol::ResetCounter, _>(|zelf, req| {
-                use protocol::reset_counter::*;
-                // NOTE: Currently, we only handle "local resets" for port 0,
-                // the "self" port.
-                if req.reset_type != ResetType::Local || req.port_id != 0 {
-                    return Err(protocol::Error {
-                        code: protocol::ErrorCode::Unspecified,
-                        data: [0; 4],
-                    });
-                }
-
-                Ok(ResetCounterResponse {
-                    count: zelf.opts.reset.resets_since_power_on() as u16,
-                })
-            })
-            .handle::<protocol::DeviceUptime, _>(|zelf, req| {
-                use protocol::device_uptime::*;
-                // NOTE: CUrrently, we only handle port 0, the "self" port.
-                if req.port_id != 0 {
-                    return Err(protocol::Error {
-                        code: protocol::ErrorCode::Unspecified,
-                        data: [0; 4],
-                    });
-                }
-                Ok(DeviceUptimeResponse {
-                    uptime: zelf.opts.reset.uptime(),
-                })
-            })
-            .handle::<protocol::RequestCounter, _>(|zelf, _| {
-                use protocol::request_counter::*;
-                Ok(RequestCounterResponse {
-                    ok_count: zelf.ok_count,
-                    err_count: zelf.err_count,
-                })
-            })
-            .run(self, host_port, arena);
-
-        match result {
-            Ok(_) => self.ok_count += 1,
-            Err(_) => self.err_count += 1,
-        }
-        result
-    }
-
-    /// Start and process a outgoing request.
-    ///
-    /// The request message will be read from `req`, while the response
-    /// message will be written to `resp`.
-    #[cfg_attr(test, inline(never))]
-    pub fn process_response<'req>(
-        &mut self,
-        _device_port: &mut dyn net::DevicePort,
-        _arena: &'req impl Arena,
-    ) -> Result<(), Error> {
-        unimplemented!()
+        self.rot.process_request(host_port, arena)
     }
 }
 
 #[cfg(test)]
 mod test {
     use super::*;
-    use core::time::Duration;
-
     use crate::crypto::ring;
     use crate::hardware::fake;
     use crate::hardware::Identity as _;
     use crate::io::Cursor;
     use crate::mem::BumpArena;
     use crate::net::DevicePort;
+    use crate::protocol;
     use crate::protocol::capabilities::*;
+    use crate::protocol::device_id;
     use crate::protocol::wire::FromWire;
     use crate::protocol::wire::ToWire;
     use crate::protocol::Header;
+    use core::time::Duration;
```

Comment:
Looks like this wandered away from the separate block it was in? In general I've tried to keep imports from different crates in different stanzas.